packages/console/src/pages/ConnectorDetails/components/ConnectorContent.tsx

```diff
@@ -42,9 +45,25 @@ const ConnectorContent = ({ isDeleted, connectorData, onConnectorUpdated }: Prop
   } = methods;
 
   useEffect(() => {
-    const { name, logo, logoDark, target } = connectorData.metadata;
-    const { config, syncProfile } = connectorData;
+    const { formItems, metadata, config, syncProfile } = connectorData;
+    const { name, logo, logoDark, target } = metadata;
+
+    const configFormData = new Map(
+      formItems
+        ? formItems.map((item) => {
+            const value = config[item.key] ?? item.defaultValue;
+
+            if (item.type === ConnectorConfigFormItemType.Json) {
+              return [item.key, JSON.stringify(value, null, 2)];
+            }
+
+            return [item.key, value];
+          })
+        : []
+    );
+
     reset({
+      ...Object.fromEntries(configFormData),
```

Comment:
should configFormData and the original config both exist in the form data?

Reply:
Yes, the "config form items" is flattend